flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartitionFactory.java

```diff
@@ -131,8 +132,15 @@ public ResultPartition create(
 				bufferCompressor,
 				bufferPoolFactory);
 
-			for (int i = 0; i < subpartitions.length; i++) {
-				subpartitions[i] = new PipelinedSubpartition(i, pipelinedPartition);
+			if (type == ResultPartitionType.PIPELINED_APPROXIMATE) {
```

Comment:
If we expect more partitions type, I'd rather see some factory lambda being extracted and used.

```
				BiFunction<Integer, PipelinedResultPartition, PipelinedSubpartition> factory;
				if (type == ResultPartitionType.PIPELINED_APPROXIMATE) {
					factory = PipelinedApproximateSubpartition::new;
				} else {
					factory = PipelinedSubpartition::new;
				}

				for (int i = 0; i < subpartitions.length; i++) {
					subpartitions[i] = factory.apply(i, pipelinedPartition);
				}
```